src/main/metabase/api.ts

```diff
@@ -176,6 +202,52 @@ export async function addDataSources(): Promise<Array<Record<string, unknown>>>
   ];
 }
 
+export async function buildNewSQLPayload(tableName: string, aidColumns: string[]): Promise<InitialQueryPayloads> {
+  const databaseId = await getAnonymizedAccessDbId();
+  const tableId = await getTableId(databaseId, tableName);
+  const fields = (await get(`api/table/${tableId}/query_metadata`)).fields as Array<{ id: number; name: string }>;
+
+  // Picks some fields which aren't AIDs to put in the GROUP BY.
+  const nonAidField = fields.find((field) => !aidColumns.includes(field.name));
+
+  const columnSQL = nonAidField ? `"${nonAidField.name}",` : '';
+  const groupBySQL = nonAidField ? 'GROUP BY 1' : '';
+  const query = `SELECT ${columnSQL} count(*) FROM "${tableName}" ${groupBySQL}`;
+
+  const breakout = nonAidField ? [['field', nonAidField.id, null]] : [];
+
+  const sqlPayload = {
```

Comment:
Can you split the part for making the SQL query object into a function that accepts the db_id and query string only?
Will help me in the future to not rediscover the syntax.